provider/virtio_mem_utils.py

```diff
@@ -81,19 +96,57 @@ def check_memory_devices(device_id, cfg_req_size_param, threshold, vm, test):
             size = int(data.get("size"))
             node = int(data.get("node"))
             requested_size = int(data.get("requested-size"))
-            error_context.context("node: %d, id: %s, size: %d and requested-size: %d"
-                                  % (node, id, size, requested_size), test.log.debug)
+            error_context.context(
+                "node: %d, id: %s, size: %d and requested-size: %d"
+                % (node, id, size, requested_size),
+                test.log.debug,
+            )
             node_size = float(get_node_size(node, vm, test))
-            normalized_size = float(normalize_data_size(str(size), 'M'))
+            normalized_size = float(normalize_data_size(str(size), "M"))
             mem_limit = (node_size - normalized_size) * threshold
             if size != requested_size and normalized_size > mem_limit:
-                test.fail("Error, size: %d and requested-size: %d are not equals."
-                          % (size, requested_size))
+                test.fail(
+                    "Error, size: %d and requested-size: %d are not equals."
+                    % (size, requested_size)
+                )
             if cfg_req_size_param != "0":
-                req_size_param = int(float(normalize_data_size(cfg_req_size_param, 'B')))
+                req_size_param = int(
+                    float(normalize_data_size(cfg_req_size_param, "B"))
+                )
             else:
                 req_size_param = int(cfg_req_size_param)
             error_context.context("req_size_param: %d" % req_size_param, test.log.debug)
             if requested_size != req_size_param:
-                test.fail("Error, requested-size: %d is not the specified in the cfg: %d."
-                          % (requested_size, req_size_param))
+                test.fail(
+                    "Error, requested-size: %d is not the specified in the cfg: %d."
+                    % (requested_size, req_size_param)
+                )
+
+
+def count_memslots(vm, mem_object_id):
+    """
+    Returns the number of memslots assigned to the current virtio-mem device
+    :param vm: the VM object
+    :param mem_object_id: the ID of the memory object device
+    """
+    output = vm.monitor.info("mtree").splitlines()
+    memslots = set()
+    for memory_region in output:
+        memory_region = memory_region.strip()
+        if mem_object_id in memory_region and "memslot" in memory_region:
+            memslots.add(memory_region)
+    return len(memslots)
```

Comment:
I think we could use a more simple way to get the number of the mem slots by using `re.findall`
like:
```suggestion
    output = vm.monitor.info("mtree")
    return len(re.findall(r"%s*memslot" % mem_object_id), re.MULTILINE) # The regular expression just a example
```
Anyway, your solution is good to me.

Reply:
Thanks for the suggestion!, code updated now (including the missing output variable in findall function)

Reply:
Hi @mcasquer 
There is a bit wrong with my suggestion `return len(re.findall(r"%s*memslot" % mem_object_id), re.MULTILINE)`, the `re.MULTILINE` should be `return len(re.findall(r"%s*memslot" % mem_object_id, output, re.MULTILINE))`.
Sorry for that.
And the regular expression about the `r"%s*memslot" % mem_object_id`, it just an example, please make it base on the actual content. Thanks.

Reply:
@YongxueHong ok, there is one extra thing to consider and it's to avoid the duplicates, so also including a `set()` for this purpose